corehq/apps/change_feed/consumer/feed.py

```diff
@@ -174,9 +174,10 @@ def _filter_partitions(self, topic_partitions):
             if self.process_num == 0:
                 return None
             else:
+                num_processes = self.num_processes - 1
                 return [
-                    topic_partitions[num::self.num_processes]
-                    for num in range(self.num_processes - 1)
+                    topic_partitions[num::num_processes]
+                    for num in range(num_processes)
                 ][self.process_num - 1]
```

Comment:
If I want to have 16 data processes and 1 migration process, is `num_processes` 16 or 17? I think I know the answer but I just want to double check my understanding. If the answer is 17, is the migration process "process 0" and the data processes processes 1 through 16?

Reply:
> If I want to have 16 data processes and 1 migration process, is `num_processes` 16 or 17? I think I know the answer but I just want to double check my understanding. If the answer is 17, is the migration process "process 0" and the data processes processes 1 through 16?

Correct, the answer is 17 and the migration process is process 0.

Reply:
One thing I don't know is whether this process will get continually re-spawned by https://github.com/dimagi/commcare-hq/blob/579b52f1037d30c8e9184214f7278bb1d6a8b9d2/corehq/ex-submodules/pillowtop/run_pillowtop.py#L37-L40

Reply:
@snopoke Yeah I think the idea is that the dedication migration process runs in a loop, where "run" is these lines: https://github.com/dimagi/commcare-hq/blob/21154a505544c66f85733ade83051afc620ecf5e/corehq/ex-submodules/pillowtop/pillow/interface.py#L110-L114, which basically checks if there are any "boostrap"/migrations to do, if there are does them, sleeps 10 seconds and then exits. So it's basically a heartbeat to see if there's any work to do and most of the time it does nothing, sleeps 10 seconds, and tries again.